src/__tests__/actions-test.ts

```diff
@@ -305,71 +300,81 @@ describe("actions", () => {
 
   describe("editStages", () => {
     it("dispatches request and success", async () => {
+      const registrationUrl = "/admin/libraries/registration";
       const dispatch = stub();
       const formData = new (window as any).FormData();
       formData.append("registration_stage", "production");
 
-      const fetchMock = stub().returns(new Promise<any>((update, reject) => {
-        update({ status: 200 });
-      }));
-      fetch = fetchMock;
+      fetchMock
+        .mock(registrationUrl, "done");
 
       await actions.editStages(formData)(dispatch);
-      expect(dispatch.callCount).to.equal(2);
+      let fetchArgs = fetchMock.calls();
+
+      expect(dispatch.callCount).to.equal(3);
```

Comment:
This had to be updated to `3` but the title of this test is `request and success`, not `request and success and load`. This also seems wrong to me... Does posting data to edit a stage also return data to load?

For the function itself, in `postForm` there was a `response.text()` function but it did not resolve. Since it did not resolve, the tests kept timing out. I'm not sure if it's right for this mocking function to assume that there is value/function for `.text()`.

Reply:
I don't know what's supposed to happen, but it does seem questionable that you had to change this.

Reply:
I'm not sure why it was `2` before but the server returns the `uuid` after the success dispatch in the `response.text()` method. So then the dispatch calls `.load()`. `3` seems to be the right number of calls for this.